railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -184,6 +187,10 @@ def initialize(*args)
         if !options[:skip_active_record] && !DATABASES.include?(options[:database])
           raise Error, "Invalid value for --database option. Supported for preconfiguration are: #{DATABASES.join(", ")}."
         end
+
+        # Force sprockets to be skipped when generating http only app.
+        # Can't modify options hash as it's frozen by default.
+        self.options = options.merge(skip_sprockets: true, skip_javascript: true).freeze if options[:api]
```

Comment:
I'm suggesting to compose the conditionals in new methods. This way you would have to change the templates, but we don't need to change an implementation detail.